docs/reference/modules/migration.md

```diff
@@ -0,0 +1,117 @@
+<!-- Some are used within the module and would just create noise here. -->
+
+Because this module has an alias, you can call these from another module from the alias path. For example, `self.apos.migration.add()`.
+
+### `add(name, fn)`
+
+Add a migration function (`fn`) to be invoked when the `@apostrophecms/migration:migrate` task is invoked. Each migration is only invoked once, however they will all be invoked on a brand-new site, so they must check whether a change is actually needed before executing changes. They must also be harmless to run twice.
+
+Functions receive no arguments, but have access to the modules where they are registered (via `self`) and the Apostrophe object (`self.apos`).
+
+<AposCodeBlock>
+```javascript
+module.exports = {
+  extend: '@apostrophecms/piece-type'
+  init(self) {
+    // 👇 Adding a data migration related to this module.
+    self.apos.migration.add('fix-roses', self.paintRosesRed);
+  },
+  methods(self) {
+    return {
+      // 👇 Registering a method to run in the migration.
+      async painRosesRed () {
+        const req = self.apos.task.getReq('editor');
+
+        await self.apos.migration.eachDoc({
+          type: 'rose'
+        }, async (doc) => {
+          if (doc.color === 'white') {
+            doc.color = 'red';
+            await self.update(req, doc);
+          }
+        });
+      }
+    }
+  }
+};
+```
+  <template v-slot:caption>
+    modules/rose/index.js
+  </template>
+</AposCodeBlock>
+
+::: warning
+If running multiple instances of the website on a server, note that previous instances of the site are not stopped automatically while migrations are run. Migrations must minimize their impact on currently running instances of older versions of the site. It is safer to copy data to new properties and to not *remove* existing data until a subsequent migration.
+
+<!-- TODO: Link to the global module reference page here. -->
+If you absolutely must prevent requests from being executed during the migration, wrap them with the `await apos.global.busy(fn)` API. Note that this API involves a significant startup delay to allow existing requests to terminate.
+:::
+
+### `async eachDoc(criteria, limit, iterator)`
+
+Invokes the `iterator` function once for each doc in the `aposDocs` collection. The iterator function receives the document as an argument and is run as an `async` function. This method will never visit the same doc twice in a single call, even if modifications are made.
+
+The `criteria` object is used to find documents to process, formatted the same way as an argument to the [query builder of the same name](/reference/query-builders.md#criteria). `limit` should be an integer and the number of documents to process in parallel, though it may be omitted. If only two arguments are passed in, `limit` is assumed to be 1 (only one doc may be processed at a time).
```

Comment:
The criteria object is a mongodb criteria object passed directly to mongodb. (So is that query builder, but it's important to emphasize the raw mongo-ness of these methods.)